packages/runtime/container-runtime/src/opLifecycle/README.md

```diff
@@ -66,47 +68,30 @@ As `FlushMode.TurnBased` accumulates ops, it is the most vulnerable to run into
-
-See [below](#how-grouped-batching-works) for an example.
-
-### Risks
+Compressing a batch yields a batch of the same size, but only the first message has any content - the compressed payload containing all the original batch's content.
+The rest of the batch's messages are empty placeholders to reserve sequence numbers for the compressed messages.
 
-This option should **ONLY** be enabled after observing that 99.9% of your application sessions contains these changes (runtime version "2.0.0-internal.7.0.0" or later). Containers created with this option may not open in future versions of the framework.
-
-This option will change a couple of expectations around message structure and runtime layer expectations. Only enable this option after testing
-and verifying that the following expectation changes won't have any effects:
+## Grouped batching
 
--   batch messages observed at the runtime layer will not match messages seen at the loader layer (i.e. grouped form at loader layer, ungrouped form at runtime layer)
--   messages within the same batch will have the same sequence number
--   client sequence numbers on batch messages can only be used to order messages with the same sequenceNumber
--   requires all ops to be processed by runtime layer (version "2.0.0-internal.1.2.0" or later https://github.com/microsoft/FluidFramework/pull/11832)
+With Grouped Batching enabled (it's on by default), all batch messages are combined under a single "grouped" message _before compression_. Upon receiving this new "grouped" message, the batch messages will be extracted, and they each will be given the same sequence number - that of the parent "grouped" message.
 
-Grouped batching may become problematic for batches which contain reentrant ops. This is the case when changes are made to a DDS inside a DDS 'onChanged' event handler. This means that the reentrant op will have a different reference sequence number than the rest of the ops in the batch, resulting in a different view of the state of the data model.
+The purpose for enabling grouped batching before compression is to eliminate the empty placeholder messages in the chunks. Without grouped batching, a batch with many messages (i.e. more than 4k) will go over the batch size limit just on empty op envelopes alone.
 
-Therefore, when grouped batching is enabled, all batches with reentrant ops are rebased to the current reference sequence number and resubmitted to the data stores so that all ops are in agreement about the state of the data model and ensure eventual consistency.
+Grouped batching is only relevant for `FlushMode.TurnBased`, since `OpGroupingManagerConfig.opCountThreshold` defaults to 2. Grouped batching is opaque to the server and implementations of the Fluid protocol do not need to alter their behavior to support this client feature.
```

Comment:
FlushMode are not exposed as API, so maybe not worth mentioning? Or mention in a way that is not confusing to people who do not know anything about it and can't find it.

Reply:
Well, these are dev docs for people working in the code, not for consumers, IMO.  So I think it's fine.

Reply:
Actually, FlushMode is `@alpha`, you can pass it in on `IContainerRuntimeOptions`